selfservice/flow/settings/hook.go

```diff
@@ -223,6 +226,12 @@ func (e *HookExecutor) PostSettingsHook(w http.ResponseWriter, r *http.Request,
 		return nil
 	}
 
-	x.ContentNegotiationRedirection(w, r, ctxUpdate.GetIdentityToUpdate().CopyWithoutCredentials(), e.d.Writer(), returnTo.String())
+	i, ok := ctxUpdate.GetIdentityToUpdate()
```

Comment:
Could you please add a test for these checks? :)

Reply:
This check is useless because there is `ctxUpdate.UpdateIdentity(i)` at [line 166](https://github.com/ory/kratos/blob/1f9bd1c1951eb48a35be384c6a06a866992be9de/selfservice/flow/settings/hook.go#L166). So I removed these.